<DatePicker/> - expose theming capabilities

sivanharel opened the pull request:
<!---
Thanks for submitting a pull request 😄 !
-->

<!---
- Be as descriptive as possible when explaining what was changed.
- Link to an issue if one exists
-->

### 🔦 Summary

...

<!--- Please mark all checkbox. If one is not relevant - delete it -->

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers <!--- Please tag the relevant team member -->
- [ ] 👨‍🎨 UX change is approved by the Design System UX <!--- Please tag the relevant team member -->
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

WixBuildServer commented:
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6195

sivanharel commented:
@shlomitc , @EgoziE, @jonathanadler 
Hi, I would like to share with you the technical difficulties that were raised when I tried to theme the DatePicker component. Since the WSR Calendar component is using the react-day-picker (an external component) all the css were overridden by the global (it's not stylable classes). In addition, since we couldn't add data attributes such as data-hook for these elements, the drivers are using the classes as the selectors.

In order to theme this component, I changed all the classes to be stylable classes bypassing them in the prop: "classNames". (Calender.js lines: 300-320)

After I changed the classes all the tests were failed. Since it uses the old classes. The problem is that I can't import the new stylable classes in the driver files so I can't use the stylable classes instead.
So I thought that I could keep both classes: the old ones for the tests and the new ones for the styling. (although it is not the optimal way it looked like the easiest). This also caused a lot of problems, the react-day-picker component cannot handle multiple classes. When I passed to the 'outside modifier' these two classes:` outside: `outside ${classes.outsideDay}`` I got a JS runtime error from the react-day-picker component that the element does not exist. (Helper.js file, function getDayNode, line 136)
`const selector = `.${dayQuery}:not(.${outsideDayQuery})`` 
it happens since the `outsideDayQuery` is multiple classes in our case.

I also tried to use other selectors in the drivers like the aria-disabled but every change that I did cause other issues. I am trying to find the best solution and it seems like I am spending a lot of time on it. Jonathan tried to help me as well and we sat together for more than an hour and וnfortunately we didn't find a solution each time that we fixed something it caused another issue. 

Maybe you guys, @shlomitc , @EgoziE , have another idea for how to solve it?

shlomitc commented on src/Calendar/Calendar.st.css:
```diff
@@ -20,50 +25,62 @@
-.calendar :global(.DayPicker) {
+.container {
   font-size: 14px;
   line-height: 18px;
   font-family: var(--wsr-font-family, "HelveticaNeueW01-45Ligh", "HelveticaNeueW02-45Ligh", "HelveticaNeueW10-45Ligh",
-      "Helvetica Neue", "Helvetica", "Arial", "メイリオ, meiryo", "ヒラギノ角ゴ pro w3", "hiragino kaku gothic pro", "sans-serif");
+  "Helvetica Neue", "Helvetica", "Arial", "メイリオ, meiryo", "ヒラギノ角ゴ pro w3", "hiragino kaku gothic pro", "sans-serif");
```
can this be taken from the `Foundation/stylable/typography.st.css` definitions instead?

shlomitc reviewed (commented):
excellent job!

sivanharel commented on src/Calendar/Calendar.js:
```diff
@@ -270,6 +271,9 @@ export default class Calendar extends React.PureComponent {
     const selectedDays = this._getSelectedDays(value);
 
+    let containerClass = `DayPicker ${classes.calendarContainer}`;
+    containerClass += numOfMonths > 1 ? ` ${classes.twoMonths}` : '';
```
It is not the updated code.
I changed the class `twoMonths` for a stylable state and add it to the relevant element which is the element with the months class.

After talking with Tom David and Barak,
I used the stylable `cssStates` function in order to define Stylble states instead of concatenating multiple classes.
It's a nice solution :)

[link for the relevant code](https://github.com/wix/wix-style-react/pull/6195/files#diff-2b89bf2a982d50fb7ed2347659011d7eca3c0e03c5defa6e37605d387b69d28dR320)
